### PR TITLE
ARM platforms: Add support for SGI575

### PR DESCRIPTION
**Pull request by npoushin**
Add support for System Guidance for Infrastructure platform SGI575.

Change-Id: I0125c2ed4469fbc8367dafcc8adce770b6b3147d
Signed-off-by: Nariman Poushin <nariman.poushin@linaro.org>

### CODE REVIEW

**Comment by npoushin** [comment]
Refactored the code to leave only sgi575 folder under plat/arm/board, moving the sgi common code under plat/arm/css

File: plat/arm/css/sgi/include/platform_def.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+#define CSS_SGI_DEVICE_BASE	(0x20000000)
+#define CSS_SGI_DEVICE_SIZE	(0x20000000)
+#define CSS_SGI_MAP_DEVICE	MAP_REGION_FLAT(		\
+					CSS_SGI_DEVICE_BASE,	\
+					CSS_SGI_DEVICE_SIZE,	\
+					MT_DEVICE | MT_RW | MT_SECURE)
+
+#define PLAT_CSS_SCP_COM_SHARED_MEM_BASE	0x45400000
+#define SGI_BOOT_CFG_ADDR		0x45410000
```
Comment (antonio-nino-diaz-arm) [inline plat/arm/css/sgi/include/platform_def.h]: nit: Align this value.

File: plat/arm/css/sgi/include/sgi_plat_config.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+} css_topology_t;
+
+typedef struct css_plat_config {
+	const gicv3_driver_data_t *gic_data;
+	const css_inteconn_config_t *inteconn;
+	const css_topology_t *topology;
+} css_plat_config_t;
+
+void plat_config_init(void);
+css_plat_config_t *get_plat_config(void);
```
Comment (antonio-nino-diaz-arm) [inline plat/arm/css/sgi/include/sgi_plat_config.h]: nit: Leave an empty line before the `#endif`.

File: plat/arm/css/sgi/sgi_gic_config.c [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <sgi_plat_config.h>
+
+void plat_arm_gic_driver_init(void)
+{
+	/*
+	 * The GICv3 driver is initialized in EL3 and does not need
+	 * to be initialized again in SEL1. This is because the S-EL1
```
Comment (antonio-nino-diaz-arm) [inline plat/arm/css/sgi/sgi_gic_config.c]: nit: s/SEL1/S-EL1

File: plat/arm/css/sgi/include/platform_def.h (hunk 1 above)
Comment (npoushin) [reply plat/arm/css/sgi/include/platform_def.h]: Done

File: plat/arm/css/sgi/include/sgi_plat_config.h (hunk 2 above)
Comment (npoushin) [reply plat/arm/css/sgi/include/sgi_plat_config.h]: Done

File: plat/arm/css/sgi/sgi_gic_config.c (hunk 3 above)
Comment (npoushin) [reply plat/arm/css/sgi/sgi_gic_config.c]: Done

File: plat/arm/css/sgi/include/platform_def.h [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+#define CSS_SGI_MAX_CORES_PER_CLUSTER	4
+
+/* CPU topology */
+#define PLAT_ARM_CLUSTER_COUNT		2
+#define PLATFORM_CORE_COUNT		(PLAT_ARM_CLUSTER_COUNT *	\
+					CSS_SGI_MAX_CORES_PER_CLUSTER)
+
+#if ARM_BOARD_OPTIMISE_MEM
+
+#if IMAGE_BL31 || IMAGE_BL32
```
Comment (antonio-nino-diaz-arm) [inline plat/arm/css/sgi/include/platform_def.h]: This should be `if defined(IMAGE_BLxx)`

Comment (npoushin) [reply plat/arm/css/sgi/include/platform_def.h]: Done

**Pull request merged**